queries/flights.py

```diff
@@ -0,0 +1,502 @@
+    9: "níunda",
+    10: "tíunda",
+    11: "ellefta",
+    12: "tólfta",
+    13: "þrettánda",
+    14: "fjórtánda",
+    15: "fimmtánda",
+    16: "sextánda",
+    17: "sautjánda",
+    18: "átjánda",
+    19: "nítjánda",
+    20: "tuttugasta",
+    21: "tuttugasta og fyrsta",
+    22: "tuttugasta og annan",
+    23: "tuttugasta og þriðja",
+    24: "tuttugasta og fjórða",
+    25: "tuttugasta og fimmta",
+    26: "tuttugasta og sjötta",
+    27: "tuttugasta og sjöunda",
+    28: "tuttugasta og áttunda",
+    29: "tuttugasta og níunda",
+    30: "þrítugasta",
+    31: "þrítugasta og fyrsta",
+}
+
+
+def QFlightsQuery(node, params, result):
+    # Set the query type
+    result.qtype = _FLIGHTS_QTYPE
+
+
+def QFlightsArrivalQuery(node, params, result):
+    result["departure"] = False
+
+
+def QFlightsDepartureQuery(node, params, result):
+    result["departure"] = True
+
+
+def QFlightsArrLoc(node, params, result):
+    result["to_loc"] = result._nominative
+
+
+def QFlightsDepLoc(node, params, result):
+    result["from_loc"] = result._nominative
+
+
+_ISAVIA_FLIGHTS_URL = (
+    "https://www.isavia.is/json/flight/?cargo=0&airport={0}"
+    "&dateFrom={1}&dateTo={2}&language=is&departures={3}"
+)
+
+_FLIGHTS_CACHE_TTL = 600  # seconds, ttl = 10 mins
+
+# Cache for flights either departing or arriving
+_FLIGHT_CACHE = cachetools.TTLCache(maxsize=2, ttl=_FLIGHTS_CACHE_TTL)
+
+
+def _fetch_flight_data(
+    from_date: datetime, to_date: datetime, iata_code: str, departing: bool
+) -> List[Dict[str, Any]]:
+    """
+    Fetch data on flights to/from an Icelandic airport (given with its IATA code)
+    between from_date and to_date from Isavia's JSON API.
+    """
+    date_format = "%Y-%m-%d %H:%M"
+    from_date_str = from_date.strftime(date_format)
+    to_date_str = to_date.strftime(date_format)
+
+    # Insert GET parameters
+    url: str = _ISAVIA_FLIGHTS_URL.format(
+        iata_code.upper(), from_date_str, to_date_str, str(departing).lower()
+    )
+
+    res = query_json_api(url)
+
+    # Verify result was successful
+    if not res or "Success" not in res or not res["Success"] or "Items" not in res:
+        return []
+
+    # Add result to cache
+    _FLIGHT_CACHE[departing] = res["Items"]
+    return res["Items"]
+
+
+def _filter_flight_data(
+    flights: List[Dict[str, Any]],
+    airport: str,
+    api_airport: str,
+    n: int = 1,
+) -> List[Dict[str, Any]]:
+    """
+    Narrows down list of flight data dicts for first n flights to/from the specified airport.
+    Adds flight_time and api_airport attributes to matching flights.
+    Returns the matching flights in a list.
+    """
+    flight_time: datetime
+    flight: Dict[str, Any]
+    now: datetime = datetime.now(timezone.utc)  # Timezone aware datetime
+
+    flight_info: List[Dict[str, Any]] = []
+    for flight in flights:
+        if n <= 0:
+            break
+
+        if (
+            airport == "*"
+            or airport in flight["DisplayName"].lower()
```

Comment:
Is a string "in" operation definitely OK here, as opposed to for instance `.startswith()` or a straight comparison?